src/java.base/share/classes/java/net/DatagramSocket.java

```diff
@@ -102,10 +103,142 @@
+ * datagram packets however, the default outgoing interface for sending
+ * multicast datagram should first be configured using
+ * {@link #setOption(SocketOption, Object) setOption} and
+ * {@link StandardSocketOptions#IP_MULTICAST_IF}:
+ *
+ * <pre>{@code
+ *    DatagramSocket sender = new DatagramSocket(new InetSocketAddress(0));
+ *    NetworkInterface outgoingIf = NetworkInterface.getByName("en0");
+ *    sender.setOption(StandardSocketOptions.IP_MULTICAST_IF, outgoingIf);
+ *
+ *    // optionally configure multicast TTL; the TTL defines the scope of a
+ *    // multicast datagram, for example, confining it to host local (0) or
+ *    // link local (1) etc...
+ *    int ttl = ...; // a number betwen 0 and 255
+ *    sender.setOption(StandardSocketOptions.IP_MULTICAST_TTL, ttl);
+ *
+ *    // send a packet to a multicast group
+ *    byte[] msgBytes = ...;
+ *    InetAddress mcastaddr = InetAddress.getByName("228.5.6.7");
+ *    int port = 6789;
+ *    InetSocketAddress dest = new InetSocketAddress(mcastaddr, port);
+ *    DatagramPacket hi = new DatagramPacket(msgBytes, msgBytes.length, dest);
+ *    sender.send(hi);
+ * }</pre>
+ *
+ * <p> An instance of {@code DatagramSocket} can also be used to receive
+ * multicast datagram packets. A {@code DatagramSocket} that is created
+ * with the intent of receiving multicast datagrams should be created
+ * <i>unbound</i>. Before binding the socket, {@link #setReuseAddress(boolean)
+ * setReuseAddress(true)} should be configured:
+ *
+ * <pre>{@code
+ *    DatagramSocket socket = new DatagramSocket(null); // unbound
+ *    socket.setReuseAddress(true); // set reuse address before binding
+ *    socket.bind(new InetSocketAddress(6789)); // bind
+ *
+ *    // joinGroup 228.5.6.7
+ *    InetAddress mcastaddr = InetAddress.getByName("228.5.6.7");
+ *    InetSocketAddress group = new InetSocketAddress(mcastaddr, 0);
+ *    NetworkInterface netIf = NetworkInterface.getByName("en0");
+ *    socket.joinGroup(group, netIf);
+ *    byte[] msgBytes = new byte[1024]; // up to 1024 bytes
+ *    DatagramPacket packet = new DatagramPacket(msgBytes, msgBytes.length);
+ *    socket.receive(packet);
+ *    ....
+ *    // eventually leave group
+ *    socket.leaveGroup(group, netIf);
+ * }</pre>
+ *
+ * <p><a id="PlatformDependencies"></a><b>Platform dependencies</b>
+ * <p>The multicast implementation is intended to map directly to the native
+ * multicasting facility. Consequently, the following items should be considered
+ * when developing an application that receives IP multicast datagrams:
+ * <ol>
+ *    <li> Contrarily to {@link DatagramChannel}, the constructors of {@code DatagramSocket}
+ *        do not allow to specify the {@link ProtocolFamily} that corresponds
+ *        to the address type of the multicast groups that the {@code DatagramSocket}
+ *        will join. This is similar to a {@code DatagramChannel} that would be created
+ *        using {@link DatagramChannel#open()}.
```

Comment:
I think this paragraph might be confusing to readers. I think you just want to say that DatagramSocket does not support a way to specify the PF when creating the socket. Consequently the underlying socket may not correspond to the multicast groups that the DatagramSocket attempts to join.

Reply:
Done. I have updated the PR and integrated your previous comments too.